tests/jballerina-unit-test/src/test/resources/test-src/types/tuples/tuple_rest_descriptor_test.bal

```diff
@@ -178,6 +178,30 @@ function testSubTypingWithRestDescriptorNegative() {
     assertFalse(v2 is [string, string, string, string]);
 }
 
+function testRestVariablesWithArray() {
+    int[] [...a] = [1, 2];
+    int[] _ = a;
+    assertEquality(a, <int[]>[1, 2]);
```

Comment:
Why do we need this cast?

Reply:
I think this because the CET of `[1, 2]` will be `any` without the cast then the first if of the `assertEquality` method will not be true and the test will fail. We can fix this if we r able to change the param type of `assertEquality` to `anydata`?

Reply:
Then how to use `assertEquality` for types other than `anydata`?

Reply:
If we can't change the param type it will be fine right @KavinduZoysa?

Reply:
Yes, let's keep it